IntroComponentWithSign-upForm/script.js

```diff
@@ -0,0 +1,67 @@
+const form  = document.getElementById("form")
+const firstName  = document.getElementById("firstName")
+const lastName  = document.getElementById("lastName")
+const email  = document.getElementById("email")
+const password  = document.getElementById("password")
+//show error message
+const showErrorMessage = ( input , message ) => {
+    const span = document.querySelector("."+input.id)  
+    console.log(span.innerText = message);
+}
+// Show input success message
+const showSuccessMessage = ( input ) => {
+    const span = document.querySelector("."+input.id)
+    span.innerText = "";
+}
+//get field name
+const getFieldName = (input) =>{
+    return input.name;   
+} 
+// check required fields
+const checkRequired = (inputArr) => {    
+    inputArr.forEach( (input)=>{ 
+        if ( input.value.trim() === "")
+        { 
+            showErrorMessage(input , getFieldName(input) +"  " + "cannot be empty")
+             //icon error    
+             var erIcon = document.querySelectorAll(".iconError")
+             for (var i = 0; i < erIcon.length; i++) {
+                erIcon[i].style.visibility = "visible";;
+             }    
+            // place holder email color
+           var El = getFieldName(input)      
+           if (El === "email"){
```

Comment:
Again, this is a bad practice! Give variables good descriptive names! Even if sometimes they get a bit long... It's better to know right away what variables are used for and what content they hold, than having to dig on your codebase for answers! 😄 

```suggestion
           const inputElement = getFieldName(input);

           if (inputElement === "email") {
```